starknet_py/net/full_node_client.py

```diff
@@ -101,6 +106,102 @@ async def get_block_traces(
     ) -> BlockTransactionTraces:
         raise NotImplementedError()
 
+    async def get_events(
+        self,
+        address: Hash,
+        keys: List[Hash],
+        from_block_number: Optional[Union[int, Tag]] = None,
+        from_block_hash: Optional[Union[Hash, Tag]] = None,
+        to_block_number: Optional[Union[int, Tag]] = None,
+        to_block_hash: Optional[Union[Hash, Tag]] = None,
+        follow_continuation_token: bool = False,
+        chunk_size: int = 1,
+    ) -> EventsResponse:
+        # pylint: disable=too-many-arguments
+        """.
+        :param from_block_number: Number of the block from which events searched for **start**
+                                or literals `"pending"` or `"latest"`.
+                                Mutually exclusive with ``from_block_hash`` parameter.
+        :param from_block_hash: Hash of the block from which events searched for **start**
+                                or literals `"pending"` or `"latest"`.
+                                Mutually exclusive with ``from_block_number`` parameter.
+        :param to_block_number: Number of the block from which events searched for **end**
+                                or literals `"pending"` or `"latest"`.
+                                Mutually exclusive with ``to_block_hash`` parameter.
+        :param to_block_hash: Hash of the block from which events searched for **end**
+                                or literals `"pending"` or `"latest"`.
+                                Mutually exclusive with ``to_block_number`` parameter.
+        :param address: The address of the contract that emitted the event.
+        :param keys: List of names of events that are searched for.
+        :param follow_continuation_token: Flag deciding whether all events should be collected during one function call,
+                                defaults to False.
+        :param chunk_size: Size of chunk of events returned by one ``get_events`` call, defaults to 1 (minimum).
+
+        :returns: ``EventsResponse`` dataclass containing events and optional continuation token.
+        """
+
+        def map_keys_to_bytes(keys: List[Hash]):
+            mapped_keys = []
+            for i in keys:
+                if isinstance(i, str):
+                    mapped_keys.append(bytes(i, "utf-8"))
+                else:
+                    mapped_keys.append(bytes(i))
+            return mapped_keys
```

Comment:
I don't think we should be doing these kind of conversions inside client. Keys should be just used in the form they are passed.